internal/configs/sharding/allowlist.go

```diff
@@ -0,0 +1,47 @@
+package shardingconfig
+
+import (
+	"fmt"
+
+	bls_cosi "github.com/harmony-one/harmony/crypto/bls"
+)
+
+type Allowlist struct {
+	MaxLimitPerShard int
+	BLSPublicKeys    []bls_cosi.PublicKeyWrapper
+}
+
+func _BLS(pubkeys []string) []bls_cosi.PublicKeyWrapper {
+	blsPubkeys := make([]bls_cosi.PublicKeyWrapper, len(pubkeys))
+	for i := range pubkeys {
+		if key, err := bls_cosi.WrapperPublicKeyFromString(pubkeys[i]); err != nil {
+			panic(fmt.Sprintf("invalid bls key: %d:%s error:%s", i, pubkeys[i], err.Error()))
+		} else {
+			blsPubkeys[i] = *key
+		}
+	}
+	return blsPubkeys
+}
+
+// each time to update the allowlist, it requires a hardfork.
+var mainnetAllowlist = Allowlist{
+	MaxLimitPerShard: 0,
+	BLSPublicKeys:    _BLS([]string{}),
+}
+
+var testnetAllowlist = Allowlist{
+	MaxLimitPerShard: 4,
+	BLSPublicKeys: _BLS([]string{
+		"7915b9cbae9d675af510cb252362b80ae6d68a3684bbea203bc30d2f5fda25ffcedfa3cf2a6c1d3051469379920a418d",
```

Comment:
Please add comments of validator names and addresses.

How are different shard ids in the BLS keys handled?

Reply:
It takes the intersection of the committees of a shard and AllowList.

Reply:
we can add the comments of the validator, but blskeys in the allowlist don't bind to validators. a validator who owns the key can remove it, and then add it to another validator.

Reply:
> we can add the comments of the validator, but blskeys in the allowlist don't bind to validators. a validator who owns the key can remove it, and then add it to another validator.

Sure, but I don't think most validators are sharing BLS keys so it makes sense to at least have a record in the code.

> It takes the intersection of the committees of a shard and AllowList.
To confirm, does that mean the shardID of the allow list BLS public key is inferred from the committee members of each shard?

Reply:
> To confirm, does that mean the shardID of the allow list BLS public key is inferred from the committee members of each shard?

yeah, the allowlist just records blskeys and does not care which shard it is. if a key is in the allowlist but not elected, it will not be in the committees, so it can’t be the leader.

Reply:
> To confirm, does that mean the shardID of the allow list BLS public key is inferred from the committee members of each shard?

and each shard has its committee, keys in the committee are in the same shard.